sched/pthread/pthread_condclockwait.c

```diff
@@ -104,7 +105,7 @@ int pthread_cond_clockwait(FAR pthread_cond_t *cond,
 
   else if (!abstime)
     {
-      ret = pthread_cond_wait(cond, mutex);
+      ret = nx_pthread_cond_wait(cond, mutex);
```

Comment:
why not remove enter_cancellation_point/leave_cancellation_point from this file directly

Reply:
I suggest we leave this file as is, even though this is not a cancellation point (as it is a GNU extension of POSIX). But GNU declares that this should behave exactly like pthread_cond_timedwait , which is a cancellation point.

https://www.gnu.org/software/libc/manual/html_node/Waiting-with-Explicit-Clocks.html

At some point I hope to move this file (and the rest) to libc, and call nx_pthread_cond_wait (et al) via call gate, and in this case the cancel point logic must be in this file.

Reply:
nxsem_clockwait_uninterruptible should not (and is not) a cancellation point though.

Anything with the nx-prefix should not be a cancellation point. I think the naming is like "nx-prefix means the function is a kernel API" and kernel does not use cancel points.

Reply:
Ok, why not move enter_cancellation_point/leave_cancellation_point close nxsem_clockwait_uninterruptible ?

Reply:
That would be an option yes. I think I will just drop this patch, we can make the changes when pthread_ functions are moved to libc.

Reply:
Ok.